File: Livermore/Matlab/kernel_05_tridiag_elimination.m
function kernel_05_tridiag_elimination(LEN_1D)
    %! init
    x = randn(1, LEN_1D);
    y = randn(1, LEN_1D);
    z = randn(1, LEN_1D);

    %! loop
    for ii = 2:LEN_1D
        x(ii) = z(ii) .* (y(ii) - x(ii - 1));
    end

end

File: Livermore/Python/kernel_05_tridiag_elimination.py
def kernel_05_tridiag_elimination(LEN_1D):
    #! init
    import numpy as np

    x = np.random.randn(LEN_1D);
    y = np.random.randn(LEN_1D);
    z = np.random.randn(LEN_1D);

    #! loop
    for k in range(1, LEN_1D):
        x[k] = z[k] * (y[k] - x[k-1])
